Tasks/Gradle/gradletask.ts

```diff
@@ -88,23 +89,46 @@ if (isSonarQubeEnabled) {
     gb = sqGradle.applySonarQubeCodeCoverageArguments(gb, isCodeCoverageOpted, ccTool, summaryFile);
 }
 
+if (isPMDAnalysisEnabled) {
+    console.log(tl.loc('codeAnalysis_ToolIsEnabled'), 'PMD');
+
+    var pmdInitScriptPath: string = path.join(__dirname, 'CodeAnalysis', 'pmd.gradle');
+    gb.arg(['-I', pmdInitScriptPath]);
+}
+
 gb.exec()
-    .then(function(code) {
+    .then(function (code) {
         publishTestResults(publishJUnitResults, testResultsFiles);
         publishCodeCoverage(isCodeCoverageOpted);
+        processCodeAnalysisResults();
 
-        if (isSonarQubeEnabled) {
-            sqGradle.uploadSonarQubeBuildSummary();
-        }
         tl.exit(code);
     })
-    .fail(function(err) {
+    .fail(function (err) {
         publishTestResults(publishJUnitResults, testResultsFiles);
         console.error(err.message);
         tl.debug('taskRunner fail');
         tl.exit(1);
     })
 
+function processCodeAnalysisResults() {
+
+    if (isPMDAnalysisEnabled) {
+        tl.debug('Processing code analysis results');
+        let buildOutput: BuildOutput = new BuildOutput(tl.getVariable('build.sourcesDirectory'), BuildEngine.Gradle);
+        let caOrchestrator: CodeAnalysisOrchestrator = new CodeAnalysisOrchestrator(
+            buildOutput,
+            path.join(tl.getVariable('build.artifactStagingDirectory'), ".codeAnalysis"),
+            tl.getVariable('build.buildNumber'));
+
+        caOrchestrator.orchestrateCodeAnalysisProcessing();
```

Comment:
There's a definite code smell here - the caller is checking whether PMD is enabled, and then the caOrchestrator is checking too. The orchestrator is checking to decide whether to create a PMD parser. Instead, the caller could create the parser and pass it in.